app/component/map/origin-popup.cjsx

```diff
@@ -1,93 +1,42 @@
 React            = require 'react'
 isBrowser        = window?
-Popup            = if isBrowser then require './dynamic-popup'
+Popup            = if isBrowser then require('./dynamic-popup').default
 SearchActions    = require '../../action/search-actions'
 intl             = require 'react-intl'
 Icon          = require '../icon/icon'
 
 class OriginPopup extends React.Component
-
-  constructor: ->
-    super
-    @toref = undefined
-    @state =
-      position: undefined
-      yOffset: 0
-
   @contextTypes:
     getStore: React.PropTypes.func.isRequired
     executeAction: React.PropTypes.func.isRequired
     intl: intl.intlShape.isRequired
 
   componentDidMount: =>
-    @context.getStore('EndpointStore').addChangeListener @onEndpointChange
-    @context.getStore('PositionStore').addChangeListener @onPositionChange
-    if @context.getStore('EndpointStore').isPendingPopup()
-      @showEndpoint()
-
-  componentWillUnmount: =>
-    @context.getStore('EndpointStore').removeChangeListener @onEndpointChange
-    @context.getStore('PositionStore').removeChangeListener @onPositionChange
+    if @props.shouldOpen
+      setImmediate @display
```

Comment:
[This](https://github.com/HSLdevcom/digitransit-ui/blob/update-dynamic-popup/app/server.cjsx#L71)?